src/keydra/clients/qualys.py

```diff
@@ -0,0 +1,142 @@
+import requests
+import xmltodict
+from collections import OrderedDict
+
+
+# https://www.qualys.com/platform-identification/
+API_URL = {
+    'US1': 'https://qualysapi.qualys.com',
+    'US2': 'https://qualysapi.qg2.apps.qualys.com',
+    'US3': 'https://qualysapi.qg3.apps.qualys.com',
+    'EU1': 'https://qualysapi.qualys.eu',
+    'EU2': 'https://qualysapi.qg2.apps.qualys.eu',
+    'IN1': 'https://qualysapi.qg1.apps.qualys.in',
+    'CA1': 'https://qualysapi.qg1.apps.qualys.ca',
+    'AE1': 'https://qualysapi.qg1.apps.qualys.ae'
+}
+
+
+class ConnectionException(Exception):
+    pass
+
+
+class PasswordChangeException(Exception):
+    pass
+
+
+class QualysClient(object):
+    def __init__(self, username, password, platform='US3', verify=True):
+        '''
+        Initializes a Qualys client
+
+        :param username: Username used to connect
+        :type username: :class:`string`
+        :param password: Password to connect
+        :type password: :class:`passwd`
+        :param platform: The Qualys platform to use
+        :type platform: :class:`string`, optional
+        :param verify: Verify TLS
+        :type verify: :class:`bool`
+        '''
+
+        self._baseurl = API_URL[platform]
+        self._user = username
+        self._pass = password
+
+        self._headers = {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        }
+
+        if type(self._user_list()) is not OrderedDict:
+            raise ConnectionException('Unknown connection error')
```

Comment:
Could we provide some more details on what's wrong, and potentially hint at how to resolve it? 🙂